crates/blobber/src/cache.rs

```diff
@@ -0,0 +1,205 @@
+use crate::{BlobFetcherError, Blobs, FetchResult};
+use alloy::consensus::{SidecarCoder, SimpleCoder, Transaction as _};
+use alloy::primitives::{keccak256, Bytes, B256};
+use reth::transaction_pool::TransactionPool;
+use reth::{network::cache::LruMap, primitives::Receipt};
+use signet_extract::ExtractedEvent;
+use signet_zenith::Zenith::BlockSubmitted;
+use signet_zenith::ZenithBlock;
+use std::{
+    sync::{Arc, Mutex},
```

Comment:
we should use the tokio Mutex, not the std one. we want to yield back to the executor when acquiring a lock